models/Tool/Targeting/TargetGroup/Dao.php

```diff
@@ -83,6 +85,7 @@ public function save()
     public function delete()
     {
         $this->db->delete('targeting_target_groups', ['id' => $this->model->getId()]);
+        $this->model->dispatchEvent(new TargetGroupEvent($this->model), TargetGroupEvents::POST_DELETE);
```

Comment:
Hmm, not sure if the DAO should dispatch the events... I think it should be the responsibility of the model (like everywhere else).

Reply:
@jdreesen I agree - @kingjia90 so you can revert your last change and just add `$this->getDao()->save/delete()` to the methods instead.